servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAlivePolicies.java

```diff
@@ -0,0 +1,143 @@
+     * ping</a> if the channel is idle for the passed {@code idleDuration} and waits for {@code ackTimeout} for an ack
+     * for that <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>. Default values are used for other
+     * parameters of the returned {@link KeepAlivePolicy}.
+     *
+     * @param idleDuration {@link Duration} of idleness on a connection after which a
+     * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+     * @param ackTimeout {@link Duration} to wait for an acknowledgment of a previously sent
+     * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>.
+     * @return A {@link KeepAlivePolicy} that disables all keep alive behaviors.
+     */
+    public static KeepAlivePolicy whenIdleFor(final Duration idleDuration, final Duration ackTimeout) {
+        return new KeepAlivePolicyBuilder().idleDuration(idleDuration).ackTimeout(ackTimeout).build();
+    }
+
+    /**
+     * A builder of {@link KeepAlivePolicy}.
+     */
+    public static final class KeepAlivePolicyBuilder {
+        private Duration idleDuration = DEFAULT_IDLE_DURATION;
+        private Duration ackTimeout = DEFAULT_ACK_TIMEOUT;
+        private boolean withoutActiveStreams;
+
+        /**
+         * Set the {@link Duration} of idleness on a connection after which a
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+         * <p>
+         * <strong>Too short ping durations may cause high network traffic, implementations may enforce a minimum
+         * duration.</strong>
+         *
+         * @param idleDuration {@link Duration} of idleness on a connection after which a
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a> is sent.
+         * @return {@code this}.
+         * @see KeepAlivePolicy#idleDuration()
+         */
+        public KeepAlivePolicyBuilder idleDuration(final Duration idleDuration) {
+            if (idleDuration.getSeconds() < 10) {
+                throw new IllegalArgumentException("idleDuration: " + idleDuration + " (expected >= 10 seconds");
+            }
+            this.idleDuration = idleDuration;
+            return this;
+        }
+
+        /**
+         * Set the maximum {@link Duration} to wait for an acknowledgment of a previously sent
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>. If no acknowledgment is received, the
+         * connection will be closed.
+         *
+         * @param ackTimeout {@link Duration} to wait for an acknowledgment of a previously sent
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.7">ping</a>.
+         * @return {@code this}.
+         * @see KeepAlivePolicy#ackTimeout()
+         */
+        public KeepAlivePolicyBuilder ackTimeout(final Duration ackTimeout) {
+            this.ackTimeout = requireNonNull(ackTimeout);
```

Comment:
Should we check that this is at least 1 ms?

Reply:
We can also check at the build time that the `ackTimeout` is less than `idleDuration`.

Reply:
I would like to avoid making assumptions here as this does not have a network traffic impact. Although a good point that smaller values will cause issues with the scheduler. Let me also pass the unit to the internal `Scheduler` interface.

> We can also check at the build time that the ackTimeout is less than idleDuration.

There isn't really a relation b/w these two values.